druid/examples/game_of_life.rs

```diff
@@ -0,0 +1,171 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Game of life
+
+use druid::kurbo::BezPath;
+use druid::piet::{FontBuilder, ImageFormat, InterpolationMode, Text, TextLayoutBuilder};
+
+
+use druid::{Affine, AppLauncher, BoxConstraints, Color, Env, Event, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx, LocalizedString, PaintCtx, Point, Rect, RenderContext, Size, UpdateCtx, Widget, WindowDesc, Data};
+
+
+const POOL_SIZE: i32 = 1600;
+const CELL_COLOR: Color = Color::rgb8(0xf3 as u8, 0xf4 as u8, 8 as u8);
+
+#[derive(Clone)]
+struct AppData {
+    grid: [i32; 1600],
```

Comment:
I would also probably (caveat: all of this is imagined for some "real program") call `AppData` `Grid`, and I would maybe even completely abstract away the underlying storage, coming up with something like this:

```rust
struct Grid {
    storage: [bool; POOL_SIZE],
}

#[derive(Clone, Copy)]
struct GridPos {
    row: usize,
    col: usize,
}

impl Index<GridPos> for Grid {
    type Output = bool;

    fn index(&self, pos: GridPos) -> &Self::Output {
        let idx = pos.row * GRID_SIZE + pos.col;
        self.storage.index(idx)
    }
}

impl IndexMut<GridPos> for Grid {
    type Output = bool;

    fn index_mut(&mut self, pos: GridPos) -> &Self::Output {
        let idx = pos.row * GRID_SIZE + pos.col;
        self.storage.index_mut(idx)
    }
}
```

And then I would implement everything in terms of that?

Again, this is overkill for this problem but would be very reasonable for something of larger scope.